programs/util.h

```diff
@@ -356,9 +356,15 @@ UTIL_STATIC const char** UTIL_createFileList(const char **inputNames, unsigned i
             size_t len = strlen(inputNames[i]);
             if (buf + pos + len >= bufend) {
                 ptrdiff_t newListSize = (bufend - buf) + LIST_SIZE_INCREASE;
-                buf = (char*)realloc(buf, newListSize);
-                bufend = buf + newListSize;
-                if (!buf) return NULL;
+                buftmp = (char*)realloc(buf, newListSize);
+                if (buftmp != NULL) {
+                    buf = buftmp;
+                    bufend = buf + newListSize;
+                    if (!buf) return NULL;
+                } else {
+                    free(buf);
+                    /* And handle error */
```

Comment:
Which error should be handled here ?
`return NULL;` ? ( ping @inikep)
The patch seems incomplete here.